### PR TITLE
#trivial Implemented a shared Page#load()

### PR DESCRIPTION
**Pull request by dbernar1**
Implemented each page object's load method as a call to a parent Page#load().

Old way:
```
load () {
        const pageUrl = buildUrl(this.componentType, this.componentName, this.path);
        this.open(pageUrl);
        this.waitForComponent();
    }
```

New way: 
```
load () {
        super.load(this.component);
}
```

### CODE REVIEW

File: packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js [hunk 1]
```diff
@@ -149,8 +140,8 @@ module.exports = class Checkout extends Page {
         return this;
     }
 
-    waitForComponent () {
-        super.waitForComponent(this.component);
+    waitForComponent (component=this.component) {
```
Comment (DamianMullins) [inline packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js]: NIT

```suggestion
    waitForComponent (component = this.component) {
```

File: packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
-
     get actionComponent () { return $('[data-test-id="action-button-component"]'); }
 
     get linkComponent () { return $('[data-test-id="link-button-component"]'); }
 
     open (url) {
         super.open(url);
     }
+
+    load (type='action') {
```
Comment (DamianMullins) [inline packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js]: NIT

```suggestion
    load (type = 'action') {
```

File: packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js [hunk 3]
```diff
@@ -124,20 +123,12 @@ module.exports = class Checkout extends Page {
      * @param {String} checkout.isValid The checkout validation
      */
 
-    load (componentToWaitFor='default') {
-        const pageUrl = buildUrl(this.componentType, this.componentName, this.path);
-        this.open(pageUrl);
-
-        switch (componentToWaitFor) {
-            case 'error':
-                this.waitForErrorPageComponent();
```
Comment (DamianMullins) [inline packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js:133]: Is the method below still required? If not you can get rid of it.

Comment (dbernar1) [reply packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js:133]: load()? That one is in use.

File: packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js [hunk 4]
```diff
@@ -1,29 +1,26 @@
...
     get actionComponent () { return $('[data-test-id="action-button-component"]'); }
 
     get linkComponent () { return $('[data-test-id="link-button-component"]'); }
 
     open (url) {
         super.open(url);
     }
+
+    load (type = 'action') {
+        if ('action' === type) {
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js]: Super small thing, but this reads a bit strange to me this way round. Reads better as

```suggestion
        if (type === 'action') {
```

File: packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js (hunk 3 above)
Comment (ashleynolan) [reply packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js:133]: I think he may mean the `waitForErrorPageComponent` method?

File: packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js (hunk 4 above)
Comment (dbernar1) [reply packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js]: Sure thing. Just gonna mention [Yoda Conditions](https://en.wikipedia.org/wiki/Yoda_conditions) here, for reference.

File: packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js (hunk 3 above)
Comment (dbernar1) [reply packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js:133]: Ah. In that case - that's coming in https://github.com/dbernar1/fozzie-components/commit/d8c494330334e48ee00203b20fb0c514e2f4a175

Comment (DamianMullins) [reply packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout.component.js:133]: 👍🏻

**Review by DamianMullins — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

File: packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js (hunk 4 above)
Comment (ashleynolan) [reply packages/components/atoms/f-button/test-utils/component-objects/f-button.component.js]: I've never heard of that before! Thanks for the link 🤣

**Pull request merged**